File: January Challenge/Java/Set Mismatch.java
class Solution {
    public int[] findErrorNums(int[] nums) {
        int i = 0;
        while(i < nums.length){
            int crt = nums[i]-1;
            if(nums[i] != nums[crt]){
                int temp = nums[i];
                nums[i] = nums[crt];
                nums[crt] = temp;
            }
            else{
                i += 1;
            }
        }
        for(int j = 0; j < nums.length; j++){
            if (nums[j] != j+1){
                return new int[]{nums[j], j+1};
            }
        }
        return new int[]{};
    }
}

File: January Challenge/Python/Set mismatch.py
class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        i = 0
        ls = []
        while i < len(nums):
            c = nums[i]-1
            if nums[i] != nums[c]:
                nums[i], nums[c] = nums[c], nums[i]
            else:
                i += 1
        
        for i in range(len(nums)):
            if nums[i] != i+1:
                ls.append(nums[i])
                ls.append(i+1)
        return ls
